test/source/tests/gmail.ts

```diff
@@ -144,23 +144,13 @@ export const defineGmailTests = (testVariant: TestVariant, testWithBrowser: Test
       testWithBrowser(async (t, browser) => {
         await BrowserRecipe.setUpCommonAcct(t, browser, 'ci.tests.gmail');
         const gmailPage = await openGmailPage(t, browser);
-        /*
-      await gmailPage.type('[aria-label^="Search"]', 'encrypted email for offline decrypt');
-      await gmailPage.press('Enter'); // submit search
-      await Util.sleep(2); // wait for search results
-      */
         await gotoGmailPage(gmailPage, '/FMfcgzGkbDWztBnnCgRHzjrvmFqLtcJD');
-        const pgpBlockUrls = await gmailPage.getFramesUrls(['/chrome/elements/pgp_block.htm'], {
-          sleep: 10,
-          appearIn: 25,
-        });
-        expect(pgpBlockUrls.length).to.equal(1);
-        await gmailPage.page.setOfflineMode(true); // go offline mode
-        await gmailPage.press('Enter'); // open the message
-        const pgpBlockFrame = await gmailPage.getFrame(['pgp_block.htm']);
+        const expectedMessage = { content: ['this should decrypt even offline'], signature: 'signed', encryption: 'encrypted' };
+        await BrowserRecipe.pgpBlockCheck(t, await gmailPage.getFrame(['/chrome/elements/pgp_render_block.htm'], { timeout: 10000 }), expectedMessage);
+        await gotoGmailPage(gmailPage, '/FMfcgzGkbDRNgcQxLmkhBCKVSFwkfdvV'); // plain convo
         await gmailPage.page.setOfflineMode(true); // go offline mode
-        await pgpBlockFrame.frame.goto(pgpBlockFrame.frame.url()); // reload the frame
-        await pgpBlockFrame.waitForContent('@pgp-block-content', 'this should decrypt even offline');
+        await gmailPage.page.goBack({ waitUntil: 'load' });
```

Comment:
What was the idea behind offline decryption?
To gracefully handle the situation when the page goes offline and the user presses `Back` button to see the previously loaded (cached) message?
I reworked the test for this scenario.
Right now it works ok as the message processing result is cached in some way in `GmailElementReplacer`, there was a leftover invisible pgp block iframe, so a little patch was required as the frameId was also inadvertently cached.
https://github.com/FlowCrypt/flowcrypt-browser/pull/5069/commits/52b13ee93ce099e5a044fd4c7ce28a2ccaf62b13

Should we support signed-only messages too (where we pull `raw` message from API)?
@tomholub @sosnovsky

Reply:
> What was the idea behind offline decryption?

Gmail webpage is actually sort of a web app that can actually work offline. Like, it will remember previous emails and you can open it on an airplane and view them, kind of like outlook.

No API calls of course, and no attachments.

So when the PGP message is rendered in the message body, we would like to still be able to render it even when offline.